### DISCUSSION TITLE
FP crashes after running model on image containing æøå in filename

### DISCUSSION THREAD

**Discussion by mhoibo** [AICAN-Research/FAST-Pathology#78] [Q&A]
The image can be viewed and inference runs fine. However, at the end of inference, likely when the result is to be stored in the Project, the program crashes:
```
HDF5-DIAG: Error detected in HDF5 (1.10.6) thread 0:
  #000: C:\workspace\FAST-dependencies\build\build_output\hdf5\src\hdf5\src\H5F.c line 444 in H5Fcreate(): unable to create file
    major: File accessibilty
    minor: Unable to open file
  #001: C:\workspace\FAST-dependencies\build\build_output\hdf5\src\hdf5\src\H5Fint.c line 1509 in H5F_open(): unable to open file: time = Fri Aug 11 14:21:19 2023
, name = 'C:/Users/SomeUser/fastpathology/projects/new_release_test//results/A05_µ°σ/BACH Classification/heatmap/heatmap.hdf5', tent_flags = 13
    major: File accessibilty
    minor: Unable to open file
  #002: C:\workspace\FAST-dependencies\build\build_output\hdf5\src\hdf5\src\H5FD.c line 734 in H5FD_open(): open failed
    major: Virtual File Layer
    minor: Unable to initialize object
  #003: C:\workspace\FAST-dependencies\build\build_output\hdf5\src\hdf5\src\H5FDsec2.c line 346 in H5FD_sec2_open(): unable to open file: name = 'C:/Users/SomeUser/fastpathology/projects/new_release_test//results/A05_µ°σ/BACH Classification/heatmap/heatmap.hdf5', errno = 2, error message = 'No such file or directory', flags = 13, o_flags = 302
    major: File accessibilty
    minor: Unable to open file
```

I guess the letters `æ`, `ø`, and `å` are replaced by other symbols (see `µ°σ` in log above), but then the result stored in the Project no longer matches the name in the Project.

I checked the Project, and there the name is `A05_æøå`, so these need to match.

Observed on Windows 10 using the latest artifact.

cc: @andreped

**Comment by andreped** [comment]
@smistad

**Reply by smistad** [reply]
This happens because FAST uses std::string, and this works fine on UTF-8 systems, but on windows most functions require UTF-16, which requires us to use std::wstring. So this is a bit of a mess..

There is a discussion on this issue here: https://stackoverflow.com/questions/3257263/how-do-i-get-stl-stdstring-to-work-with-unicode-on-windows
